x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/QueryableBuiltInRolesStore.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.authz.store;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.WriteRequest;
+import org.elasticsearch.xpack.core.security.action.role.BulkRolesResponse;
+import org.elasticsearch.xpack.core.security.authz.RoleDescriptor;
+import org.elasticsearch.xpack.security.support.SecurityIndexManager;
+
+import java.util.Collection;
+
+/**
+ * Wrapper around the {@link NativeRolesStore} that provides a way to create, update and delete built-in roles.
+ */
+public final class QueryableBuiltInRolesStore {
```

Comment:
Going with a wrapper in order to avoid publicly exposing `NativeRolesStore` methods that can bypass role validation.

Reply:
I like not over-exposing `NativeRolesStore` but the wrapper is a bit confusing: it's called Queryable but the only methods it has are for modifying. I tried to come up with a clearer name, but blanking on a good one since the responsibility of this class isn't very clear. 

I think especially if we manage to avoid passing in `securityIndexManager` (see my other comment) it could be fine to just expose the native role store methods directly. Or tweak this class to have a clearer purpose, where it only has a single method `syncRoles(rolesToUpdate, rolesToDelete)` and move some of the logic from `QueryableBuiltInRolesSynchronizer` into that method. WDYT?

Reply:
> I tried to come up with a clearer name, but blanking on a good one since the responsibility of this class isn't very clear.

I'm with you. I agree the class name is confusing and not describing its purpose. 

> I think especially if we manage to avoid passing in securityIndexManager (see my other comment) it could be fine to just expose the native role store methods directly.

Thanks for the suggestion. My attempt to avoid depending on `NativeRolesStore` may not be justified in this case. In the end, the `NativeRolesStore` is responsible for persisting and querying roles from `.security` index.  I think it makes sense to leverage existing method and avoid passing `securityIndexManager`. The `syncRoles(rolesToUpdate, rolesToDelete)` is an option, but I'd like to keep that functionality is separated as it feels like it belongs to the `QueryableBuiltInRolesSynchronizer`.

Reply:
I've pushed https://github.com/elastic/elasticsearch/pull/117581/commits/21646f532f4bbe0e42021e6cb88b5360ed58f023 which:

1. Removes `QueryableBuiltInRolesStore` and directly calls `NativeRolesStore`
2. Makes `putRoles` and `deleteRoles` public and removes requirement to pass SecurityIndexManager as a parameter